x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/__mocks__/rule_execution_log_client.ts

```diff
@@ -7,16 +7,17 @@
 
 import { IRuleExecutionLogClient } from '../types';
 
+export const ruleExecutionLogClientMock = {
+  create: (): jest.Mocked<IRuleExecutionLogClient> => ({
+    find: jest.fn(),
+    findBulk: jest.fn(),
+    update: jest.fn(),
+    delete: jest.fn(),
+    logStatusChange: jest.fn(),
+    logExecutionMetric: jest.fn(),
+  }),
+};
+
 export const RuleExecutionLogClient = jest
   .fn<jest.Mocked<IRuleExecutionLogClient>, []>()
-  .mockImplementation(() => {
-    return {
-      find: jest.fn(),
-      findBulk: jest.fn(),
-      create: jest.fn(),
-      update: jest.fn(),
-      delete: jest.fn(),
-      logStatusChange: jest.fn(),
-      logExecutionMetric: jest.fn(),
-    };
-  });
+  .mockImplementation(ruleExecutionLogClientMock.create);
```

Comment:
I added `ruleExecutionLogClientMock` for consistency's sake, because single `new RuleExecutionLogClient()` among `*mock.create()` methods looks alien:

```diff
const createMockClients = () => ({
  rulesClient: rulesClientMock.create(),
  licensing: { license: licensingMock.createLicenseMock() },
  clusterClient: elasticsearchServiceMock.createScopedClusterClient(),
  savedObjectsClient: savedObjectsClientMock.create(),
- ruleExecutionLogClient: new RuleExecutionLogClient(),
+ ruleExecutionLogClient: ruleExecutionLogClientMock.create(),
  appClient: siemMock.createClient(),
});
```

And kept original `export const RuleExecutionLogClient` so that Jest's module mocking can work with it:

```js
jest.mock('../../rule_execution_log/rule_execution_log_client');
```

Reply:
> I added ruleExecutionLogClientMock for consistency's sake

Ok, imho it would be nice to change all the other mocks to using this pattern, but that'd be a different story :)
LGTM 👍